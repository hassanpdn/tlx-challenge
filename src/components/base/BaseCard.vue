<template>
      <div class="card">
            <div v-if="title || $slots.cardHeader" class="card-header">
                  <slot v-if="$slots.cardHeader" name="cardHeader"></slot>
                  <span class="title" v-else-if="title"> {{title}} </span>
            </div>
            <div class="card-body">
                  <slot/>
            </div>
      </div>
</template>

<script>
      export default {
            name: 'Card',
            props : {
                  title: {
                        type: String,
                        default: ''
                  }
            }
      }
</script>

<style>
      .card {
            position: relative;
            display: flex;
            justify-content: space-between;
            flex-direction: column;
            border-radius: 0;
            border-color: rgba(102,119,136,.2);
            min-width: 0;
            word-wrap: break-word;
            background-color: #fff;
            background-clip: border-box;
            border: 1px solid rgba(0,0,0,.125);
      }

      .card-header {
            display: flex;
            justify-content: space-between;
            border-bottom: 1px solid rgba(0,0,0,.04);
            padding: 0.75rem 1.25rem;
            margin-bottom: 0;
            font-size: var(--font-size-lg);
      }
      
      .card-header:first-child {
            border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
      }

      .card-body {
            flex: 1 1 auto;
            padding: 1.25rem;
      }
</style>