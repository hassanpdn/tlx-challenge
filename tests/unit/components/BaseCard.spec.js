import { mount } from '@vue/test-utils';
import BaseCard from '@/components/base/BaseCard';

describe('BaseCard', () => {
      test("renders component", () => {
            const wrapper = mount(BaseCard);
            expect(wrapper.element).toMatchSnapshot()
      })

      test("title has the inital value of ''", () => {
            const wrapper = mount(BaseCard);
            expect(wrapper.vm.title).toEqual('')
      })

      test("has title class if title passed", () => {
            const wrapper = mount(BaseCard, {propsData: {title: 'title'}});
            expect(wrapper.find(".title")).toBeTruthy();
      })
});
