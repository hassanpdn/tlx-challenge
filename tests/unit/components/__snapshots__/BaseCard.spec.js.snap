// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard renders component 1`] = `
<div
  class="card"
>
  <!---->
   
  <div
    class="card-body"
  />
</div>
`;
